source/common/http/http3/conn_pool.h

```diff
@@ -97,6 +98,13 @@ class ActiveClient : public MultiplexedActiveClientBase {
   uint64_t quiche_capacity_ = 100;
 };
 
+class PoolConnectResultCallback {
+public:
+  virtual ~PoolConnectResultCallback() = default;
+  virtual void onConnectSucceeded() PURE;
+  virtual void onConnectFailedWithEarlyData() PURE;
```

Comment:
This method is not called currently, right? Since it's also not really implemented (it has a TODO in the body) in the Grid, it might be worth moving to a subsequent PR.